src/_premake_init.lua

```diff
@@ -482,6 +482,13 @@
 		kind = "boolean"
 	}
 
+	api.register {
+		name = "excludefrombuild",
```

Comment:
Do we need a new API here? Can't this be done through `excludes` (`removefiles`) like vstudio?

Reply:
That wouldn't result in the file being in the "solution explorer", it literally removes the file. This API just stops the file from being built, although I'm fairly certain that this already exists somewhere.

Reply:
Oh yeah, it's a flag `flags { "ExcludeFromBuild" }`.

Reply:
This looks like the exact same code as #243 which was closed as it wasn't the desired solution to the problem.

Reply:
I was confused because in the scripts we use at work we have some
```
filter { "platforms:not x" }
    excludes { "file" }
```
for platform dependent code and it behave as you said. File are in explorer view but not built.
It's actually us that does the wrong thing here.

I'd say that leveraging file filter and the flag/boolean API would make more sense but I do not have a strong opinion either way.